test/src/066-killall/main

```diff
@@ -3,9 +3,134 @@
 cvmfs_test_name="Killing all cvmfs2 instances"
 cvmfs_test_suites="quick"
 
+SRC="$(mktemp --suffix=FROZENFS)"
+DEST=/tmp/freezeme
+MOUNTPOINT=/cvmfs
+
+fs_setup(){
+  logdebug "Creating a filesystem on ${DEST}.."
+  if [ ! -d "$DEST" ]; then
+    mkdir $DEST
+  fi
+
+  if [ ! -d "$DEST" ] || [ ! -f "$SRC" ]; then
+    exit 51
+  fi
+
+	fallocate -l 8M -- "$SRC"
+	mkfs.ext4 -q -- "$SRC"
+	sudo mount -o loop -- "$SRC" "$DEST"
+}
+
+fs_cleanup(){
+  logdebug "Cleaning up ${DEST}.."
+
+  # If /tmp/freezeme is a mountpoint
+  # unfreeze it and unmount it
+  if ! mountpoint -q "${DEST}"; then
+    sudo fsfreeze -u ${DEST} 2> /dev/null
+    sudo umount ${DEST}
+  fi
+  if [ -f ${SRC} ]; then
+    rm -- ${SRC}
+  fi
+}
+
+freeze()
+{
+  logdebug "Freezing ${DEST}.."
+	sudo fsfreeze -f -- "$DEST"
+}
+
+unfreeze() {
+  logdebug "Unfreezing ${DEST}.."
+  sudo fsfreeze -u -- "$DEST"
+}
+
+get_D_state_pid() {
+  local d_pid
+
+  for pid in $(ps -eT -o tid,comm | awk '$2=="cvmfs2" {print $1}' | xargs);
+  do
+    read -r _ _ state _ < /proc/$pid/stat
+    if [ "$state" = "D" ]; then
+      echo $pid
+      break
+    fi
+  done
+}
+
+get_fuse_devices_of_process(){
+  local pid
+  # Trim whitespaces in pid
+  pid="$(echo -e "$1" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')"
+
+  if [ -z "$pid" ]; then
+    return 50
+  fi
+  local cvmfs_devices=""
+  while read -r line; do
+    # Get the interesting fields from every line of mountinfo;
+    # those are the mountpoint and the device id
+    device_id=$(echo "$line" | cut -d' ' -f 3)
+    current_mountpoint=$(echo "$line" | cut -d' ' -f 5)
+
+    if [[ "$current_mountpoint" == "$MOUNTPOINT"/*  ]]; then
+      if ! grep -q "$device_id" <<< "$cvmfs_devices"; then
+        cvmfs_devices=$cvmfs_devices" $(echo $device_id | cut -d ':' -f 2)"
+      fi
+    fi
+  done < "/proc/$pid/mountinfo"
+
+  echo "$cvmfs_devices"
+}
+
+test_extra_args(){
+  logdebug "\tTesting cvmfs_config killall $1.."
+  arg="$1"
+  local repo="atlas.cern.ch"
+  freeze
+
+  logdebug "Mounting repositories and reloading.."
+  cvmfs_mount $repo || return 40
+  sudo cvmfs_config reload || return 41
+
+  logdebug "Freezing cvmfs.."
+  sudo cvmfs_talk -i $repo __testing_freeze_cvmfs &
+
+  local d_pid=$(get_D_state_pid)
```

Comment:
Checking the process state  will be interesting to follow up in the future, but I don't think it's strictly necessary for the test. you could just do an `ls $repo` save the PID of that and then check that it is killed after the connection abort